src/documents/Priorities.md

```diff
@@ -0,0 +1,107 @@
+  </thead>
+<tr><td>Numeric</td>
+<td>An <em>integer value</em> representing a fixed priority (not generally recommended). In general, higher numbers represent "greater priority" with whatever meaning
+that takes in the relevant context. For example, a listener with higher priority will fire before one with a lower priority, or options with higher priority 
+will merge on top of options with a lower priority</td><td>1, 10, 100</td></tr>
+
+<tr><td>Constraint</td>
+<td>Either of the strings <code>before</code> or <code>after</code> followed by the <code>namespace</code> of some other element (the "target element") of the same type, separated by a colon.
+The framework will sort all the elements in the same set so that this element will be placed immediately before or after the target element, 
+unless a further constraint positions a third listener in between them.</td><td><code>"before:bindMarkup"</code>, <code>"after:computeLayout"</code></td></tr>
+
+<tr><td>Extremal Priority</td>
+<td>Either of the strings <code>first</code> or <code>last</code>, either by themselves, or followed by an <strong>extremal priority class</strong> separated by a colon. Elements annotated <code>first</code> or
+<code>last</code> will sort either before or after all those which have been given finite numerical priorities (although a constraint-based priority will always have its directive honoured, even if
+it involves being situated beyond an element with extremal priority). If the string is followed by <code>:class</code> for one of the supported classes, this increases the "level of infinity" that the priority
+represents to that of the class, so that it will beat any extremal priority of a lower class or an unqualified extremal priority. The currently supported extremal priority classes are <code>testing</code> and <code>authoring</code>.
+So, an element listed <code>last:authoring</code> will beat an element listed <code>last:testing</code>, which will beat any element merely listed as <code>last</code>.</td>
+<td><code>"last"</code>, <code>"last:testing"</code></td>
+</table>
+
+Note that the framework's algorithm for sorting by priority uses a [stable sorting algorithm](https://en.wikipedia.org/wiki/Sorting_algorithm#Stability) so that in the absence of any priorities, any
+original ordering of the elements will be preserved. Elements without any priority will sort close to those which have been specified with an explicit numeric priority of 0, whilst retaining any original relative ordering. 
+In the case of event listeners, this will be significant since it will respect any procedural order of addition, as well as any order
+resulting from grade merging. For options distributions, the natural ordering will be derived instead from the tree's topology (distributions which travelled a longer distance will
+be weaker than those which travelled a shorter distance), and numeric priorities should not be used.
+
+## Example of priority resolution for listeners
```

Comment:
I would suggest putting this example *after* the "Table of supported sites for priorities"